pkg/kubelet/preemption/preemption_test.go

```diff
@@ -458,23 +458,31 @@ func admissionRequirementListEqual(list1 admissionRequirementList, list2 admissi
 	return true
 }
 
-// this checks if the lists contents contain all of the same elements.
-// this is not correct if there are duplicate pods in the list.
-// for example: podListEqual([a, a, b], [a, b, b]) will return true
+// podListEqual checks if the lists contents contain all of the same elements.
 func podListEqual(list1 []*v1.Pod, list2 []*v1.Pod) bool {
 	if len(list1) != len(list2) {
 		return false
 	}
-	for _, a := range list1 {
-		contains := false
-		for _, b := range list2 {
-			if a == b {
-				contains = true
-			}
-		}
-		if !contains {
+
+	m1 := map[*v1.Pod]int{}
+	m2 := map[*v1.Pod]int{}
```

Comment:
I would like to avoid allocating two maps.
How about this pattern:
```
for list1 {
  m[val] = m[val] + 1
}
for list2 {
  m[val] = m[val] - 1
}

for m {
  if m[val] != 0 {
    return false
  }
}
```
?